circle.yml

```diff
@@ -246,6 +246,41 @@ commands:
       - store-results:
           module_target: "libnavigation-ui"
 
+  generate-core-version-name:
+    steps:
+      - run:
+          name: Generate Core version name
+          command: |
+            if [[ $CIRCLE_TAG == release-core-.* ]]; then
+              POM_CORE_VERSION_NAME=`echo $CIRCLE_TAG | cut -d '-' -f 3`
+            elif [[ $CIRCLE_TAG == qa-release-core-.*
+                      || $CIRCLE_TAG == prod-release-core-.* ]]; then
+              POM_CORE_VERSION_NAME=`echo $CIRCLE_TAG | cut -d '-' -f 4`
+            else
+              echo "Exiting the job as this is not a release TAG"
+              circleci step halt
+            fi
+            echo "export POM_CORE_VERSION_NAME=$POM_CORE_VERSION_NAME" >> $BASH_ENV
+
+  generate-ui-version-name:
+    steps:
+      - run:
+          name: Generate UI version name
+          command: |
+            if [[ $CIRCLE_TAG == release-ui-.*-core-.* ]]; then
+              POM_UI_VERSION_NAME=`echo $CIRCLE_TAG | cut -d '-' -f 3`
+              POM_CORE_VERSION_NAME=`echo $CIRCLE_TAG | cut -d '-' -f 5`
+            elif [[ $CIRCLE_TAG == qa-release-ui-.*-core-.*
+                      || $CIRCLE_TAG == prod-release-ui-.*-core-.* ]]; then
+              POM_UI_VERSION_NAME=`echo $CIRCLE_TAG | cut -d '-' -f 4`
+              POM_CORE_VERSION_NAME=`echo $CIRCLE_TAG | cut -d '-' -f 6`
+            else
+              echo "Exiting the job as this is not a release TAG"
+              circleci step halt
```

Comment:
Looking at docs, this will skip the job, but also report success. Would `exit 1` work instead, indicating that the job failed?

Reply:
Depends. Do we completely want to stop the CI and report error or do we want to let the CI run but not publish the artifacts?

Reply:
I'd say we want to stop. If we cut a tag that doesn't fall into the above filters, it means that there's something wrong.

Reply:
But of course, we'll first need to check if the tag exists in the first place, like:
```
if [[ -n "$CIRCLE_TAG" ]]; then
```